lib/ecto/adapters/myxql.ex

```diff
@@ -312,27 +312,42 @@ defmodule Ecto.Adapters.MyXQL do
   def structure_dump(default, config) do
     table = config[:migration_source] || "schema_migrations"
     path  = config[:dump_path] || Path.join(default, "structure.sql")
+    dump_prefixes = config[:dump_prefixes] || [config[:database]]
```

Comment:
```suggestion
    prefixes = config[:dump_prefixes] || [config[:database]]
```
to be consistent with postgres and also because we are already inside the dump functions